app/controllers/katello/api/v2/subscriptions_controller.rb

```diff
@@ -77,8 +77,8 @@ def index_relation
     def show
       org_id = Organization.current&.id
       @resource = Katello::Pool.with_identifier(params[:id])
-      if @resource.organization_id != org_id
-        fail ActiveRecord::RecordNotFound, N_('This subscription is not relevant to the current organization.')
+      if @resource.organization_id != org_id && !User.current.organizations&.pluck(:id)&.include?(@resource.organization_id)
+        fail ActiveRecord::RecordNotFound, N_('This subscription is not relevant to the current user and organization.')
```

Comment:
Your change is consistent with what's here and it fixes the problem, but I am curious about what's here (cc @jeremylenz )

If try to lookup a subscription in an organization which I don't have access to I should get a 404, and that is what happens. However, I should not be informed of whether the subscription exists or not. Since I don't have access to the organization it's irrelevant. On the other hand if that subscription *is* in my organization and I lack permission to view it, that would be permission denied error (403).

@jeremylenz is the message you added here for consumption in the UI?

Reply:
@jturel Yes, this was to address the user flow of changing the org while you happen to be on the subscription details page.  The error message shows up as a Toast notification.

Reply:
See [comment](https://github.com/Katello/katello/pull/8262#issuecomment-527635305) on 25113

Reply:
Thanks for confirming!

It looks like the Pool model isn't fully integrated into our permission system:

```
[7] pry(#<Katello::Api::V2::SubscriptionsController>)> ::Katello::Pool.authorized
NoMethodError: undefined method `authorized' for #<Class:0x00007f7ff4f60c18>
from /home/vagrant/foreman/.vendor/ruby/2.5.0/gems/activerecord-5.2.1/lib/active_record/dynamic_matchers.rb:22:in `method_missing'
```

That's a separate issue and I know we have one other files for Repositories so I won't file another since I think we'll look at permissions at a high level when addressing it. That said, this change does seem to address what was reported